db/users/utils_test.go

```diff
@@ -0,0 +1,107 @@
+package users_test
```

Comment:
Is there any reason to use a dedicated package for test file?

IMO you can still use `package users` and then not export `IsUsernameValid` function.

Reply:
Good question ;)

Yes, I use a dedicated package for what is called "black-box" testing, that emphasis on user behavior testing. 
More a habit I have from my previous company ^^, and realize that the opposed "white-box" testing suits better for the CLI.

I'll readapt and keep in the same package.

Short discussion black-box vs white-box testing:
https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language/31443271#31443271